icu4c/source/common/brkeng.cpp

```diff
@@ -299,6 +306,91 @@ ICULanguageBreakFactory::loadDictionaryMatcherFor(UScriptCode script) {
     return nullptr;
 }
 
+
+URegistryKey ICULanguageBreakFactory::addExternalEngine(
+    ExternalBreakEngine* engine, UErrorCode& status) {
```

Comment:
This adopts `engine`, so it should probably delete it in case of failure. Use LocalPointers.